plugins/extensions/plugins/RocketsPlugin.cpp

```diff
@@ -176,10 +192,33 @@ void RocketsPlugin::_broadcastWebsocketMessages()
 
     if (_engine->getCamera().getModified())
     {
-        json message;
-        message["event"] = ENDPOINT_CAMERA;
-        message["data"] = _engine->getCamera().getSerializable()->toJSON();
-        _httpServer->broadcastText(message.dump());
+        const auto& message =
+            _buildJsonMessage(ENDPOINT_CAMERA,
+                              _engine->getCamera().getSerializable()->toJSON());
+        _httpServer->broadcastText(message);
+    }
+
+    if (_engine->getModified())
+    {
+        _requestProgress();
+        const auto& message =
+            _buildJsonMessage(ENDPOINT_PROGRESS, _remoteProgress.toJSON());
+        _httpServer->broadcastText(message);
+    }
+
+    if (_engine->getRenderer().haveNewImage())
```

Comment:
hmmm, shouldn't this method be called hasNewImage() instead of haveNewImage()?

Reply:
Hmm, true. I'll have to recheck also if this works with Optix.